### ISSUE TITLE
Type does not narrow for readonly record

### ISSUE THREAD

**Issue by KavinduZoysa** [ballerina-platform/ballerina-lang#34320]
**Description:**
$subject

**Steps to reproduce:**
```
import ballerina/io;
public type R readonly & record {|
    int i; // error: invalid operation: type 'record {| readonly int i; |} & readonly?' does not support field access
|}?;

function foo(R r) returns int {
    if r == () {
        return -1;
    }
    else {
        return r.i;
    }
}

public function main() {
    int i = foo({ i: 2 });
    io:println(i);
}
```
Removing `readonly` eliminate this error;

**Affected Versions:**
swan-lake beta4

**Comment by jclark** [comment]
Using parentheses also works around the problem:  `(readonly & record { ... })?`

**Comment by KavinduZoysa** [comment]
Fixed in `Ballerina 2201.3.0`
